crates/storage/provider/src/providers/static_file/writer.rs

```diff
@@ -444,11 +473,61 @@ impl StaticFileProviderRW {
         Ok(result)
     }
 
-    /// Removes the last `number` of transactions from static files.
+    /// Adds an instruction to prune transactions during commit.
     ///
-    /// # Note
-    /// Commits to the configuration file at the end.
+    /// Note: last_block refers to the block the unwinds ends at.
     pub fn prune_transactions(
+        &mut self,
+        to_delete: u64,
+        last_block: BlockNumber,
+    ) -> ProviderResult<()> {
+        debug_assert_eq!(self.writer.user_header().segment(), StaticFileSegment::Transactions);
+        self.queue_prune(to_delete, Some(last_block))
+    }
+
+    /// Adds an instruction to prune receipts during commit.
+    ///
+    /// Note: last_block refers to the block the unwinds ends at.
+    pub fn prune_receipts(
+        &mut self,
+        to_delete: u64,
+        last_block: BlockNumber,
+    ) -> ProviderResult<()> {
+        debug_assert_eq!(self.writer.user_header().segment(), StaticFileSegment::Receipts);
+        self.queue_prune(to_delete, Some(last_block))
+    }
+
+    /// Adds an instruction to prune headers during commit.
+    ///
+    /// Note: last_block refers to the block the unwinds ends at.
+    pub fn prune_headers(&mut self, to_delete: u64) -> ProviderResult<()> {
```

Comment:
can we add the explanation for `to_delete` in the doc comment and make the name consistent with the `prune_transaction_data` method argument?